api/wallet_router_integration_test.go

```diff
@@ -0,0 +1,322 @@
+	assert.NoError(t, err)
+
+	chatAddress := derivedAccounts[pathEIP1581Chat].Address
+	found = keystoreContainsFileForAccount(keyStoreDir, chatAddress)
+	require.True(t, found)
+
+	defaultSettings, err := defaultSettings(genAccInfo.KeyUID, genAccInfo.Address, derivedAccounts)
+	require.NoError(t, err)
+	nodeConfig, err := DefaultNodeConfig(defaultSettings.InstallationID, &requests.CreateAccount{
+		LogLevel: defaultSettings.LogLevel,
+	})
+	require.NoError(t, err)
+	nodeConfig.DataDir = rootDataDir
+	nodeConfig.KeyStoreDir = keyStoreDir
+
+	nodeConfig.NetworkID = 1
+	nodeConfig.HTTPEnabled = true
+	nodeConfig.WalletConfig = params.WalletConfig{
+		Enabled: true,
+	}
+	nodeConfig.UpstreamConfig = params.UpstreamRPCConfig{
+		URL:     "http://0.0.0.0:8545",
+		Enabled: true,
+	}
+	nodeConfig.Networks = []params.Network{
+		params.Network{
+			ChainID:                uint64(31337),
+			ChainName:              "Anvil",
+			DefaultRPCURL:          "http://0.0.0.0:8545",
+			RPCURL:                 "http://0.0.0.0:8545",
+			ShortName:              "eth",
+			NativeCurrencyName:     "Ether",
+			NativeCurrencySymbol:   "ETH",
+			NativeCurrencyDecimals: 18,
+			IsTest:                 false,
+			Layer:                  1,
+			Enabled:                true,
+			// RelatedChainID:         goerliChainID,
+		},
+	}
+
+	profileKeypair := &accounts.Keypair{
+		KeyUID:      genAccInfo.KeyUID,
+		Name:        "Profile Name",
+		Type:        accounts.KeypairTypeProfile,
+		DerivedFrom: masterAddress,
+	}
+
+	profileKeypair.Accounts = append(profileKeypair.Accounts, &accounts.Account{
+		Address:   types.HexToAddress(chatAddress),
+		KeyUID:    profileKeypair.KeyUID,
+		Type:      accounts.AccountTypeGenerated,
+		PublicKey: types.Hex2Bytes(accountInfo.PublicKey),
+		Path:      pathEIP1581Chat,
+		Wallet:    false,
+		Chat:      true,
+		Name:      "GeneratedAccount",
+	})
+
+	for p, dAccInfo := range derivedAccounts {
+		found = keystoreContainsFileForAccount(keyStoreDir, dAccInfo.Address)
+		require.NoError(t, err)
+		require.True(t, found)
+
+		if p == pathDefaultWalletAccount {
+			wAcc := &accounts.Account{
+				Address: types.HexToAddress(dAccInfo.Address),
+				KeyUID:  genAccInfo.KeyUID,
+				Wallet:  false,
+				Chat:    false,
+				Type:    accounts.AccountTypeGenerated,
+				Path:    p,
+				Name:    "derivacc" + p,
+				Hidden:  false,
+				Removed: false,
+			}
+			if p == pathDefaultWalletAccount {
+				wAcc.Wallet = true
+			}
+			profileKeypair.Accounts = append(profileKeypair.Accounts, wAcc)
+		}
+	}
+
+	account := multiaccounts.Account{
+		Name:      profileKeypair.Name,
+		Timestamp: 1,
+		KeyUID:    profileKeypair.KeyUID,
+	}
+
+	err = backend.ensureAppDBOpened(account, password)
+	require.NoError(t, err)
+
+	err = backend.StartNodeWithAccountAndInitialConfig(account, password, *defaultSettings, nodeConfig, profileKeypair.Accounts, nil)
+	require.NoError(t, err)
+
+	return backend, stop, nil
```

Comment:
Same here for `stop`, using `t.Cleanup` help

Reply:
Updated.